### PR TITLE
add function find to scene-graph

### PR DESCRIPTION
**Pull request by JoneLau**

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -148,6 +150,56 @@ function enabledInHierarchy(node, includeSelf = true) {
...
+ * @method find
+ * @param {Node} root
+ * @param {string} path
+ */
+function find(root, path) {
+  let node = null;
+  let names = path.split('/');
+
+  if (names[0] != root.name)
+    return node;
```
Comment (jwu) [inline lib/utils.js]:
```
if () {
   ...
}
```

File: lib/utils.js [hunk 2]
```diff
@@ -148,6 +150,56 @@ function enabledInHierarchy(node, includeSelf = true) {
...
+/**
+ * @method find
+ * @param {Node} root
+ * @param {string} path
+ */
+function find(root, path) {
+  let node = null;
+  let names = path.split('/');
+
+  if (names[0] != root.name)
```
Comment (jwu) [inline lib/utils.js]: `!=` => `!==`

Comment (jwu) [inline lib/utils.js]: 请按我们的规范标准书写:

```
if () {
   ...
}
```

File: lib/utils.js [hunk 3]
```diff
@@ -148,6 +150,56 @@ function enabledInHierarchy(node, includeSelf = true) {
...
+    return node;
+
+  walk(root, (child, parent, level) => {
+    if (names[level] !== child.name || node != null) {
```
Comment (jwu) [inline lib/utils.js]: 1!=1 => 1!==1

**Comment by JoneLau** [comment]
update cocos-creator/engine-3d#152

**Pull request merged**